tests/returntypes/test_parameter_shift_hessian_new.py

```diff
@@ -0,0 +1,1757 @@
+
+        tapes, fn = qml.gradients.param_shift_hessian(tape)
+        hessian = fn(qml.execute(tapes, dev, gradient_fn=None))
+
+        assert isinstance(hessian, tuple)
+        assert len(hessian) == 2
+
+        for hess, exp in zip(hessian, expected):
+            assert isinstance(hess, tuple)
+            assert len(hess) == 2
+            assert all(isinstance(h, tuple) for h in hess)
+            assert all(len(h) == 2 for h in hess)
+            assert all(
+                all(isinstance(h_, np.ndarray) and h_.shape == exp[0][0].shape for h_ in h)
+                for h in hess
+            )
+            assert np.allclose(hess, exp)
+
+    def test_multi_probs_multi_params(self):
+        """Test that the correct hessian is calculated for a tape with multiple operators
+        and multiple probability outputs"""
+        dev = qml.device("default.qubit", wires=2)
+
+        x = np.array([0.1, 0.4], requires_grad=True)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RY(x[0], wires=0)
+            qml.RY(x[1], wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.probs(wires=[1])
+            qml.probs(wires=[0, 1])
+
+        a = [
+            np.cos(x[0] / 2) ** 2,
+            np.sin(x[0] / 2) ** 2,
+            np.cos(x[1] / 2) ** 2,
+            np.sin(x[1] / 2) ** 2,
+        ]
+        expected = (
+            (
+                (
+                    0.5 * np.cos(x[0]) * np.cos(x[1]) * np.array([-1, 1]),
+                    0.5 * np.sin(x[0]) * np.sin(x[1]) * np.array([1, -1]),
+                ),
+                (
+                    0.5 * np.sin(x[0]) * np.sin(x[1]) * np.array([1, -1]),
+                    0.5 * np.cos(x[0]) * np.cos(x[1]) * np.array([-1, 1]),
+                ),
+            ),
+            (
+                (
+                    0.5 * np.cos(x[0]) * np.array([-a[2], -a[3], a[3], a[2]]),
+                    0.25 * np.sin(x[0]) * np.sin(x[1]) * np.array([1, -1, 1, -1]),
+                ),
+                (
+                    0.25 * np.sin(x[0]) * np.sin(x[1]) * np.array([1, -1, 1, -1]),
+                    0.5 * np.cos(x[1]) * np.array([-a[0], a[0], a[1], -a[1]]),
+                ),
+            ),
+        )
+
+        tapes, fn = qml.gradients.param_shift_hessian(tape)
+        hessian = fn(qml.execute(tapes, dev, gradient_fn=None))
+
+        assert isinstance(hessian, tuple)
+        assert len(hessian) == 2
+
+        for hess, exp in zip(hessian, expected):
+            assert isinstance(hess, tuple)
+            assert len(hess) == 2
+            assert all(isinstance(h, tuple) for h in hess)
+            assert all(len(h) == 2 for h in hess)
+            assert all(
+                all(isinstance(h_, np.ndarray) and h_.shape == exp[0][0].shape for h_ in h)
+                for h in hess
+            )
+            assert np.allclose(hess, exp)
+
+
+@pytest.mark.skip("Requires QNode integration for new return types")
+class TestParameterShiftHessianQNode:
```

Comment:
The remainder of these tests (copied from the original file) require QNode integration.